docs/source/developers/docker.rst

```diff
@@ -0,0 +1,143 @@
+.. raw:: html
+
+   <!--
+   Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+   -->
+
+Running Docker Builds
+=====================
+
+Most of our Linux based continuous integration tasks are decoupled from public
+CI services using docker and docker-compose. Keeping the CI configuration
+minimal makes local reproducibility possible.
+
+Usage
+-----
+
+There are multiple ways to execute the docker based builds. The recommented is
+to use the archery tool:
+
+Installation:
+
+.. code:: bash
+
+    pip install -e dev/archery[docker]
+
+Execute a build:
+
+.. code::bash
+
+    archery docker run conda-python
```

Comment:
Would I ever call `archery docker` and not immediately put `run` next? If not, why not just `archery run conda-python`?

Reply:
I kept that open for features like listing available build(er)s. I can change this to something like `archery build conda-python` if you find it more convenient.

Reply:
Sounds like YAGNI to me, anyone else have an opinion?

Reply:
Just added the `archery docker images` command. We can refine the CLI API later and even keep the old commands if we choose to.